dtfs-central-api/src/helpers/map-correction-reasons-to-formatted-values.ts

```
@@ -1,35 +1,53 @@
-import { FeeRecordEntity, RECORD_CORRECTION_REASON, RecordCorrectionReason, getFormattedMonetaryValue } from '@ukef/dtfs2-common';
+import {
+  FeeRecordCorrectionEntity,
+  RecordCorrectionValues,
+  RECORD_CORRECTION_REASON,
+  RecordCorrectionReason,
+  getFormattedMonetaryValue,
+} from '@ukef/dtfs2-common';
 
 /**
- * returns the value of the fee record property that corresponds to the reason
+ * returns the value of the provided correction values that corresponds to the reason
  * or '-' if the reason is RECORD_CORRECTION_REASON.OTHER
- * @param feeRecord - the fee record
+ * @param correctedValues - the previousValues or correctedValues
  * @param reason - the reason for the correction
- * @returns - string with the value of the fee record property that corresponds to the reason
+ * @returns - string with the value of the previous or corrected values that corresponds to the reason
  */
-export const getFormattedOldValueForCorrectionReason = (feeRecord: FeeRecordEntity, reason: RecordCorrectionReason) => {
+export const getFormattedCorrectValueForCorrectionReason = (correctedValues: RecordCorrectionValues, reason: RecordCorrectionReason) => {
   switch (reason) {
     case RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT:
-      return feeRecord.facilityId;
+      return correctedValues.facilityId;
     case RECORD_CORRECTION_REASON.REPORTED_CURRENCY_INCORRECT:
-      return feeRecord.feesPaidToUkefForThePeriodCurrency;
+      return correctedValues.feesPaidToUkefForThePeriodCurrency;
     case RECORD_CORRECTION_REASON.REPORTED_FEE_INCORRECT:
-      return getFormattedMonetaryValue(feeRecord.feesPaidToUkefForThePeriod);
+      return getFormattedMonetaryValue(Number(correctedValues.feesPaidToUkefForThePeriod));
```

Why do we need `Number` here?

---

The types for these are feesPaidToUkefForThePeriod: number | null;
  feesPaidToUkefForThePeriodCurrency: Currency | null; whereas getFormattedMonetaryValue only takes number so this fixes the linting issue

---

I think we should throw an error if null, otherwise this just returns '0.00' but nothing was changed at all if it was null.

---

if nothing was changed it would not enter this case

---

Could the existing `validateRequiredFormDataField` helper be extracted into a central-api helper (vs living in a controller) eg `validateRequiredCorrectionField` and this added to each branch?

I'm doing something similar to this in my PR #4178, except this takes in an object of `RecordCorrectionValues` and reason(s) [here](https://github.com/UK-Export-Finance/dtfs2/pull/4178/files#diff-8d6531e73038ba693e5c02fc47073ab81c71796080b5020aeb8dbf130ffaefdf) so think we might have to commonise something